packages/core/src/common/test/test.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (C) 2018 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { expect } from 'chai';
+
+export async function throwsAsync(promise: Promise<any>, ...args: any[]): Promise<void> {
```

Comment:
`args` should be a RegExp, right?

Reply:
Please add a better name, for instance `expectThrows` and it is not necessarily a `Promise`.

Reply:
Well, `expect(fn).throws(params)` has 2 signatures based on the following interface (extracted from chai's `index.d.js`):
```ts
    interface Throw {
        (expected?: string|RegExp, message?: string): Assertion;
        (constructor: Error|Function, expected?: string|RegExp, message?: string): Assertion;
    }
```

chai's `.throws` can indeed take a RegEx, but it will test the regex against the error message. Which is not what we want if we want to test the type of Error (usually simply `Error`).

But when trying to implement the overloads to replace the `...args: any[]` I couldn't get it to compile.

Secondly this `throwsAsync` is useful only when monitoring async functions (or regular functions that return a promise, because async functions return a promise too). The StackOverflow version was testing a call on an async function (`await fn()`) but the problem is that most of Theia's async functions takes parameters. And at the end of the day the thing that is being awaited is a Promise. Maybe there are other awaitable types I don't know of ?

I could rename it to `test.expectThrowsAsync` ?

Reply:
@lmcbout